File: app/src/main/java/com/lucas/omnia/activities/BlocksActivity.java
package com.lucas.omnia.activities;

import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import com.lucas.omnia.R;
import com.lucas.omnia.adapters.BlockAdapter;
import com.lucas.omnia.models.Block;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class BlocksActivity extends BaseActivity {

    private RecyclerView recyclerView;
    private static BlocksActivity instance;
    private RequestQueue requestQueue;
    private static Context ctx;
    private final String URL = "https://dadosabertos.camara.leg.br/api/v2/blocos";

    public BlocksActivity() {}

    private BlocksActivity(Context context) {
        ctx = context;
        requestQueue = getRequestQueue();
    }

    public static synchronized BlocksActivity getInstance(Context context) {
        if (instance == null) {
            instance = new BlocksActivity(context);
        }
        return instance;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_blocks);

        setProgressBar(R.id.blocks_pb);
        showProgressBar();

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);

        recyclerView = findViewById(R.id.blocks_rv);
        recyclerView.setLayoutManager(layoutManager);

        showProgressBar();
        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
                (Request.Method.GET, URL, null, response -> {
                    List<Block> blockList = new ArrayList<>();
                    try {
                        JSONArray jsonArray = response.getJSONArray("dados");

                        for (int i = 0; i < jsonArray.length(); i++) {
                            JSONObject jsonObject = jsonArray.getJSONObject(i);
                            Block block = new Block(jsonObject.getString("nome"));
                            blockList.add(block);
                        }

                        final BlockAdapter blockAdapter = new BlockAdapter(blockList);
                        recyclerView.setAdapter(blockAdapter);
                        hideProgressBar();
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                },
                        error -> {
                    Toast.makeText(this, "Server error", Toast.LENGTH_SHORT).show();
                    hideProgressBar();
                });
        hideProgressBar();

        // Access the RequestQueue through your singleton class.
        BlocksActivity.getInstance(this).addToRequestQueue(jsonObjectRequest);
    }

    public RequestQueue getRequestQueue() {
        if (requestQueue == null) {
            // getApplicationContext() is key, it keeps you from leaking the
            // Activity or BroadcastReceiver if someone passes one in.
            requestQueue = Volley.newRequestQueue(ctx.getApplicationContext());
        }
        return requestQueue;
    }

    public <T> void addToRequestQueue(Request<T> req) {
        getRequestQueue().add(req);
    }
}

File: app/src/main/java/com/lucas/omnia/activities/BlocksActivity.kt
package com.lucas.omnia.activities

import android.content.Context
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.VolleyError
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.lucas.omnia.R
import com.lucas.omnia.adapters.BlockAdapter
import com.lucas.omnia.models.Block
import org.json.JSONException
import org.json.JSONObject
import java.util.*

class BlocksActivity : BaseActivity {
    private var requestQueue: RequestQueue? = null

    constructor()
    private constructor(context: Context) {
        ctx = context
        requestQueue = getRequestQueue()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_blocks)
        setProgressBar(R.id.blocks_pb)
        showProgressBar()
        val layoutManager = LinearLayoutManager(this)
        val recyclerView: RecyclerView? = findViewById(R.id.blocks_rv)
        recyclerView?.layoutManager = layoutManager
        showProgressBar()
        val jsonObjectRequest = JsonObjectRequest(Request.Method.GET, BLOCKS_URL, null, { response:
                                                                                 JSONObject ->
            val blockList: MutableList<Block> = ArrayList()
            try {
                val jsonArray = response.getJSONArray("dados")
                for (i in 0 until jsonArray.length()) {
                    val jsonObject = jsonArray.getJSONObject(i)
                    val block = Block(jsonObject.getString("nome"))
                    blockList.add(block)
                }
                val blockAdapter = BlockAdapter(blockList)
                recyclerView?.adapter = blockAdapter
                hideProgressBar()
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        }
        ) { error: VolleyError? ->
            Toast.makeText(this, "Server error: $error", Toast.LENGTH_SHORT).show()
            hideProgressBar()
        }
        hideProgressBar()

        // Access the RequestQueue through your singleton class.
        getInstance(this)!!.addToRequestQueue(jsonObjectRequest)
    }

    private fun getRequestQueue(): RequestQueue? {
        if (requestQueue == null) {
            // getApplicationContext() is key, it keeps you from leaking the
            // Activity or BroadcastReceiver if someone passes one in.
            requestQueue = Volley.newRequestQueue(ctx!!.applicationContext)
        }
        return requestQueue
    }

    fun <T> addToRequestQueue(req: Request<T>?) {
        getRequestQueue()!!.add(req)
    }

    companion object {
        const val BLOCKS_URL = "https://dadosabertos.camara.leg.br/api/v2/blocos"

        private var instance: BlocksActivity? = null
        private var ctx: Context? = null
        @Synchronized
        fun getInstance(context: Context): BlocksActivity? {
            if (instance == null) {
                instance = BlocksActivity(context)
            }
            return instance
        }
    }
}